README.md

```diff
@@ -1,2 +1,36 @@
+Therefore the worst case running time of the algorithm can be expressed as a quadratic or more specifically **_O_(n^2)**
+
+## (c)
+Changing line 4 will change how the algorithm checks the list but not the result of it.
+Originally it will run every item in the list by using the variable **i** against the items in the same list by using the variable **j**.
+After the change it will run every item in the list using variable **i** against every item that variable **i** has already passed through by using variable **j**. By using this method it will check through all of the elements in the list, but by checking only those that **i** has already been to, ending in **i** reaching the end of the list and checking against **j** which would be one index from the end of the list.
+
+## (d)
+The algorithm will take a longer time to run the further into the list it is checking, this is because as **i** increases, so does the amount of indices that **j** will check, this will increase linearly, but will since it starts with such a small amount of indeces to iterate through, the time can be averaged as if it checked half of the list each time, we can infer that it runs in half the time and therefore half of the time
+
+## (e)
+The first _for loop_, represented as **i**, has not changed and is still a linear equation. The second _for loop_, represented by **j** however, has been changed; the time taken by **j** has a much lower average time as a constant than before, but it relies on where **i** is in the list and therefore will still increase in time as **n** increases, and is therefore still a linear search. Using the method I showed in part (b), I can assume that the product of the two linear searches is still a quadratic, although it will get exponentially faster than the algorithm in part (b) as **n** increases.
+
+## (f)
+The [Python documentation of time complexity for various functions](https://wiki.python.org/moin/TimeComplexity) gives the time complexity of the _sort_ function as **_O_(n _log_ n)
+
+## (g)
+First of all, the algorithm calls the sort method which has a notation of **_O_(n _log_ n)**
+The second part of the algorithm then has a search based on the length of the list which as previously explained has a notation of **_0_(n)**.
+We can therefore assume that the full time complexity would be **(n _log_ n) * (n)**, simplified to **(n^2 _log_ n^2)**. Meaning that the proper notation of the algorithm is **_O_(_log_ n^2)**.
```

Comment:
The sorting and search happen one after the other, so the time complexity is added together: `O(n log n + n)`. Which simplifies to `O(n log n)`, since `n log n` grows faster than `n` when `n` is large.